cdc/kv/store_op.go

```diff
@@ -34,26 +33,24 @@ func GetSnapshotMeta(tiStore tidbkv.Storage, ts uint64) (*meta.Meta, error) {
 	return meta.NewSnapshotMeta(snapshot), nil
 }
 
-// CreateTiStore creates a new tikv storage client
+// CreateTiStore creates a tikv storage client
+// Note: It will return a same storage if the urls connect to a same pd cluster,
+// so must be careful when you call storage.Close().
 func CreateTiStore(urls string, credential *security.Credential) (tidbkv.Storage, error) {
 	urlv, err := flags.NewURLsValue(urls)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	// Ignore error if it is already registered.
-	_ = store.Register("tikv", driver.TiKVDriver{})
-
-	if credential.CAPath != "" {
-		conf := tidbconfig.GetGlobalConfig()
-		conf.Security.ClusterSSLCA = credential.CAPath
-		conf.Security.ClusterSSLCert = credential.CertPath
-		conf.Security.ClusterSSLKey = credential.KeyPath
-		tidbconfig.StoreGlobalConfig(conf)
-	}
-
 	tiPath := fmt.Sprintf("tikv://%s?disableGC=true", urlv.HostString())
-	tiStore, err := store.New(tiPath)
+	securityCfg := tikvconfig.Security{
+		ClusterSSLCA:    credential.CAPath,
+		ClusterSSLCert:  credential.CertPath,
+		ClusterSSLKey:   credential.KeyPath,
+		ClusterVerifyCN: credential.CertAllowedCN,
+	}
+	d := driver.TiKVDriver{}
```

Comment:
could you add some comments about the advantage  of this refactor

Reply:
sure

Reply:
done